server/src/main/java/org/opensearch/index/search/stats/SearchStats.java

```diff
@@ -86,9 +88,19 @@ public long getTimeInMillis() {
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeVLong(current);
+            if (current < 0) {
+                logger.warn("current is negative: {}", current);
```

Comment:
nit: This message will look very ambiguous in the log file, as it does not give much context. Can you make it more verbose?

Reply:
The `logger` is declared in `PhaseStatsLongHolder` so the log prefix will provide context, but I have also expanded the message.
```
        private static final Logger logger = LogManager.getLogger(PhaseStatsLongHolder.class);
```

Unfortunately, the `PhaseStatsLongHolder` class does not have access to the specific search phase for which the statistics are being written.